src/ImageManipulation.php

```diff
@@ -1057,4 +1062,98 @@ protected function castDimension($value, $dimension)
         }
         return $value;
     }
+
+    private function copyImageBackend()
+    {
+        // Store result in new DBFile instance
+        /** @var DBFile $file */
+        $file = DBField::create_field(
+            'DBFile',
+            [
+                'Filename' => $this->getFilename(),
+                'Hash' => $this->getHash(),
+                'Variant' => $this->getVariant()
+            ]
+        );
+
+        $file->setImageBackend($this->getImageBackend());
+        return $file->setOriginal($this);
+    }
+
+    public function setAttribute($name, $value)
+    {
+        $file = $this->copyImageBackend();
+        $file->setMutableAttributes(array_merge(
+            $this->attributes,
+            [$name => $value]
+        ));
+
+        return $file;
+    }
+
+    public function setMutableAttributes($attributes)
```

Comment:
This allows us to bypass our attributes Immutability. It's admittedly a bit hacky. Maybe it should be called something like `initAttributes` to make it's purpose clearer and to make sure people understand it should only be called once.

Reply:
I'm kind of surprised we need to do this given File objects are already chaining within templates?

Reply:
The problem is that when I create a new DBFile instance, I don't have a way to copy the attributes to the new instance:
- I can't call `setAttribute` on it because I've overriwritten that method already to be immutable (I would just end up in an infinite recursion)
- I can't called the protected `$attributes` class property because `Image` and `DBFile` are not the same class and won't have access to each others' protected members.